R/weight.R

```diff
@@ -41,33 +43,216 @@ weight <- function (x) {
     currentWeight <- ShojiEntity(crGET(shojiURL(x, "fragments", "preferences")))$weight
     if (!identical(value, currentWeight)) {
         crPATCH(shojiURL(x, "fragments", "preferences"),
-            body=toJSON(list(weight=value)))
+            body = toJSON(list(weight = value)))
         x <- refresh(x)
     }
     return(x)
 }
 
 #' @rdname weight
 #' @export
-setMethod("weightVariables", "CrunchDataset",
-    function (x) weightVariables(allVariables(x)))
+is.weight <- function(x){
+    if (is.variable(x)) {
+        ds <- loadDataset(datasetReference(self(x)))
```

Comment:
There is a `datasetReference` method for `CrunchVariable` (https://github.com/Crunch-io/rcrunch/blob/master/R/variable.R#L164) so you don't need the `self()` here (and below where the same pattern is followed)